packages/core/admin/ee/admin/pages/SettingsPage/pages/ReviewWorkflows/pages/CreateView/CreateView.js

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { useFormik, Form, FormikProvider } from 'formik';
+import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { useMutation } from 'react-query';
+
+import {
+  CheckPagePermissions,
+  useAPIErrorHandler,
+  useFetchClient,
+  useNotification,
+} from '@strapi/helper-plugin';
+import { Button, Flex, Loader } from '@strapi/design-system';
+import { Check } from '@strapi/icons';
+
+import { useModels } from '../../../../../../../../admin/src/hooks';
+import { WorkflowAttributes } from '../../components/WorkflowAttributes';
+import { Stages } from '../../components/Stages';
+import { reducer, initialState } from '../../reducer';
+import { REDUX_NAMESPACE } from '../../constants';
+import { resetWorkflow } from '../../actions';
+import { useInjectReducer } from '../../../../../../../../admin/src/hooks/useInjectReducer';
+import { getWorkflowValidationSchema } from '../../utils/getWorkflowValidationSchema';
+import adminPermissions from '../../../../../../../../admin/src/permissions';
+
+import * as Layout from '../../components/Layout';
+
+export function ReviewWorkflowsCreateView() {
+  const { formatMessage } = useIntl();
+  const { post } = useFetchClient();
+  const { formatAPIError } = useAPIErrorHandler();
+  const dispatch = useDispatch();
+  const toggleNotification = useNotification();
+  const { collectionTypes, singleTypes, isLoading: isLoadingModels } = useModels();
+  const {
+    clientState: {
+      currentWorkflow: { data: currentWorkflow, isDirty: currentWorkflowIsDirty },
+    },
+  } = useSelector((state) => state?.[REDUX_NAMESPACE] ?? initialState);
+
+  const { mutateAsync, isLoading } = useMutation(
+    async ({ workflow }) => {
+      const {
+        data: { data },
+      } = await post(`/admin/review-workflows/workflow`, {
+        data: workflow,
+      });
+
+      return data;
+    },
+    {
+      onSuccess() {
+        toggleNotification({
+          type: 'success',
+          message: { id: 'notification.success.saved', defaultMessage: 'Saved' },
+        });
+      },
+    }
+  );
+
+  const createWorkflow = async (workflow) => {
+    try {
+      const res = await mutateAsync({ workflow });
+
+      return res;
+    } catch (error) {
+      toggleNotification({
+        type: 'warning',
+        message: formatAPIError(error),
+      });
+
+      return null;
+    }
+  };
```

Comment:
Let's discuss that :) The problem with axios is, that axios throws in case the API returned an error, which causes our app to crash, because `mutateAsync` does not handle rejected promises. This makes `onError` mostly unusable for us in these scenarios - hence why the error is handled in `.catch`.

Reply:
> axios throws in case the API returned an error, which causes our app to crash, because mutateAsync does not handle rejected promises.

would `mutate` mean the errors / success are handled correctly? If so what do you think pros / cons are of using each? 🤔 I'm not sure i see real benefits personally – but i've not used it much!

Reply:
I'll create a topic for the next FE sync if you don't mind and build an isolated prototype to understand the differences.

Reply:
I mean reading this article from one of the maintainers – https://tkdodo.eu/blog/mastering-mutations-in-react-query#mutate-or-mutateasync it seems like there really isn't a reason for us to be using `mutateAsync`?

Reply:
Great, thanks. Let's have this discussion outside of this PR. I'll prepare something for the next FE sync.